pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1032,7 +1032,7 @@ async def on_package(pkg):
                 Args:
                     pkg (createrepo_c.Package): A completed createrepo_c package.
                 """
-                if self.mirror and pkg.location_base or ".." in pkg.location_href:
+                if self.mirror and (pkg.location_base or ".." in pkg.location_href):
```

Comment:
This is, technically, a separate bugfix.  But the urlencode test is broken without it, and the bug only exists in master.